go-controller/pkg/ovn/base_secondary_layer2_network_controller.go

```diff
@@ -34,7 +34,9 @@ func (oc *BaseSecondaryLayer2NetworkController) stop() {
 	if oc.ipamClaimsHandler != nil {
 		oc.watchFactory.RemoveIPAMClaimsHandler(oc.ipamClaimsHandler)
 	}
-	if oc.policyHandler != nil {
+	if oc.policyHandler != nil && util.IsNetworkSegmentationSupportEnabled() && oc.IsPrimaryNetwork() {
```

Comment:
nit: just checking `oc.policyHandler != nil` should be enough. As already mentioned, you need a different handler for multinetpols

Reply:
yes, done.